src/services/expensesService.js

```diff
@@ -0,0 +1,68 @@
+const { generateUniqueId } = require('../utils/getRandomId');
+
+let expenses = [];
+
+const start = () => {
+  expenses = [];
+};
+
+const getExpenses = (
+  userId = null,
+  categories = null,
+  from = null,
+  to = null,
+) => {
+  const filteredExpenses = expenses.filter((expense) => {
+    let isValid = true;
+
+    if (userId) {
+      isValid = isValid && expense.userId === Number(userId);
+    }
+
+    if (categories && categories.length > 0) {
+      isValid = isValid && categories.includes(expense.category);
+    }
+
+    if (from && to) {
+      isValid = isValid && expense.spentAt >= from && expense.spentAt <= to;
+    } else if (from) {
+      isValid = isValid && expense.spentAt >= from;
+    } else if (to) {
+      isValid = isValid && expense.spentAt <= to;
```

Comment:
The date comparison logic assumes that `spentAt`, `from`, and `to` are comparable directly. Ensure that these values are Date objects or comparable timestamps to avoid incorrect filtering.